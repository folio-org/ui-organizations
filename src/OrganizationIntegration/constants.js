export const EDI_CODE_TYPES = [
  { label: '31B (US-SAN)', value: '31B/US-SAN' },
  { label: '014 (EAN)', value: '014/EAN' },
  { label: '091 (Supplier-assigned ID)', value: '091/Vendor-assigned' },
  { label: '092 (Library-assigned ID)', value: '092/Customer-assigned' },
];

export const FTP_TYPES = [
  { label: 'SFTP', value: 'SFTP' },
  { label: 'FTP', value: 'FTP' },
];

export const TRANSMISSION_MODES = [
  { label: 'ASCII', value: 'ASCII' },
  { label: 'Binary', value: 'Binary' },
];

export const CONNECTION_MODES = [
  { label: 'Active', value: 'Active' },
  { label: 'Passive', value: 'Passive' },
];

export const SCHEDULE_PERIODS = {
  hours: 'HOUR',
  days: 'DAY',
  weeks: 'WEEK',
  months: 'MONTH',
};

export const WEEKDAYS = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];

export const EDI_NAMING_TOKENS = {
  organizationCode: 'organizationCode',
  integrationName: 'integrationName',
  exportJobEndDate: 'exportJobEndDate',
  jobID: 'jobID',
  numberSequence: 'numberSequence',
};
