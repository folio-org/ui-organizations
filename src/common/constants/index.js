export * from './api';
export * from './categories';
export * from './events';
export * from './interfaces';
export * from './organization';
export * from './organizationTypes';
export * from './routes';
export * from './sections';
export * from './vendorCategories';
