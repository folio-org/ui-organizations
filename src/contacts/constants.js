import React from 'react';
import { FormattedMessage } from 'react-intl';

export const CONTACT_PERSON_ACCORDIONS = {
  NAME: 'contact-name',
  EMAILS: 'contact-emails',
  PHONES: 'contact-phones',
  URLS: 'contact-urls',
  ADDRESSES: 'contact-addresses',
};

export const CONTACT_PERSON_ACCORDION_LABELS = {
  [CONTACT_PERSON_ACCORDIONS.NAME]: <FormattedMessage id="ui-organizations.contactPeople.name" />,
  [CONTACT_PERSON_ACCORDIONS.EMAILS]: <FormattedMessage id="ui-organizations.contactPeople.emails" />,
  [CONTACT_PERSON_ACCORDIONS.PHONES]: <FormattedMessage id="ui-organizations.contactPeople.phoneNumbers" />,
  [CONTACT_PERSON_ACCORDIONS.URLS]: <FormattedMessage id="ui-organizations.contactPeople.urls" />,
  [CONTACT_PERSON_ACCORDIONS.ADDRESSES]: <FormattedMessage id="ui-organizations.contactPeople.addresses" />,
};

export const PRIVILEGED_CONTACT_URL_PATH = 'privileged-contacts';
