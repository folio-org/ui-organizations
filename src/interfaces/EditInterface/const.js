export const formatDD = [
  { label: '', value: '' },
  { label: 'Counter', value: 'Counter' },
  { label: 'Delimited', value: 'delimited' },
  { label: 'Excel', value: 'excel' },
  { label: 'PDF', value: 'pdf' },
  { label: 'ASCII', value: 'ascii' },
  { label: 'HTML', value: 'html' },
  { label: 'Other', value: 'other' },
];

export const deliveryMethodDD = [
  { label: '', value: '' },
  { label: 'Online', value: 'Online' },
  { label: 'FTP', value: 'FTP' },
  { label: 'Email', value: 'Email' },
  { label: 'Other', value: 'Other' },
];
