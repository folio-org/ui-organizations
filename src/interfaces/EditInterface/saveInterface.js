export default function saveInterface(
  { vendorInterface, interfaceCredentials, interfaceId, interfaceOrg },
  { username, password, ...values },
  org,
  showCallout,
  creds = {},
) {
  const isNew = !values.id;
  const httpMethod = isNew ? 'POST' : 'PUT';
  let savedInterfaceId = values.id;

  return vendorInterface[httpMethod](values)
    .then(
      ({ id }) => {
        showCallout({ messageId: 'ui-organizations.interface.message.saved.success' });
        savedInterfaceId = id;
        if (isNew && org?.id) {
          return interfaceOrg.PUT({
            ...org,
            interfaces: [...org.interfaces, id],
          });
        }

        return undefined;
      },
      () => {
        showCallout({ messageId: 'ui-organizations.interface.message.saved.fail', type: 'error' });
      },
    )
    .catch(() => {
      showCallout({ messageId: 'ui-organizations.save.error.assignInterface', type: 'error' });
    })
    .then(() => {
      if (
        savedInterfaceId && (
          (!creds.id && (username || password)) ||
          creds.username !== username ||
          creds.password !== password
        )
      ) {
        interfaceId.replace(savedInterfaceId);

        return interfaceCredentials[creds.id ? 'PUT' : 'POST']({
          ...creds,
          username: username || '',
          password: password || '',
          interfaceId: savedInterfaceId,
        });
      }

      return savedInterfaceId;
    })
    .then(
      () => savedInterfaceId,
      () => {
        showCallout({ messageId: 'ui-organizations.save.error.creds', type: 'error' });

        return savedInterfaceId;
      },
    );
}
