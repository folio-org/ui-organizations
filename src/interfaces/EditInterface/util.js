import { INTERFACE_TYPES } from '../../common/constants';

export const getSelectOptions = (arrayOfStringValues) => arrayOfStringValues.map(type => ({
  value: type,
  label: type,
}));

export const INTERFACE_OPTIONS = getSelectOptions(INTERFACE_TYPES);
