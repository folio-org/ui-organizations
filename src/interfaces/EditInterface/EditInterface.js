import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Field } from 'react-final-form';
import { useHistory } from 'react-router';

import stripesForm from '@folio/stripes/final-form';
import {
  AppIcon,
  IfPermission,
} from '@folio/stripes/core';
import {
  Button,
  Checkbox,
  checkScope,
  Col,
  HasCommand,
  Headline,
  Pane,
  PaneMenu,
  Row,
  Select,
  TextArea,
  TextField,
} from '@folio/stripes/components';

import {
  handleKeyCommand,
  validateRequired,
  validateURL,
} from '@folio/stripes-acq-components';

import { ORGANIZATIONS_ROUTE } from '../../common/constants';
import TogglePassword from '../../Utils/TogglePassword';
import { INTERFACE_OPTIONS } from './util';

const EditInterface = ({
  handleSubmit,
  onClose,
  paneTitle,
  pristine,
  submitting,
  formatDD,
  deliveryMethodDD,
}) => {
  const history = useHistory();
  const getLastMenu = () => (
    <PaneMenu>
      <FormattedMessage id="ui-organizations.interface.button.save">
        {(title) => (
          <Button
            buttonStyle="primary"
            disabled={pristine || submitting}
            marginBottom0
            onClick={handleSubmit}
            title={title}
            type="submit"
          >
            {title}
          </Button>
        )}
      </FormattedMessage>
    </PaneMenu>
  );

  const shortcuts = [
    {
      name: 'cancel',
      shortcut: 'esc',
      handler: handleKeyCommand(onClose),
    },
    {
      name: 'save',
      handler: handleKeyCommand(handleSubmit, { disabled: pristine || submitting }),
    },
    {
      name: 'search',
      handler: handleKeyCommand(() => history.push(ORGANIZATIONS_ROUTE)),
    },
  ];

  return (
    <HasCommand
      commands={shortcuts}
      isWithinScope={checkScope}
      scope={document.body}
    >
      <Pane
        appIcon={
          <AppIcon
            app="organizations"
            appIconKey="organizations"
          />
        }
        defaultWidth="fill"
        dismissible
        id="edit-interface"
        lastMenu={getLastMenu()}
        onClose={onClose}
        paneTitle={paneTitle}
      >
        <Row>
          <Col
            xs={12}
            md={8}
            mdOffset={2}
          >
            <Row>
              <Col xs={6}>
                <Field
                  component={Select}
                  dataOptions={INTERFACE_OPTIONS}
                  fullWidth
                  label={<FormattedMessage id="ui-organizations.interface.type" />}
                  multiple
                  name="type"
                  type="select"
                />
              </Col>
            </Row>
            <Row>
              <Col
                xs={12}
                md={6}
              >
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.name" />}
                  name="name"
                  required
                  validate={validateRequired}
                  component={TextField}
                  fullWidth
                />
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.uri" />}
                  name="uri"
                  type="url"
                  validate={validateURL}
                  component={TextField}
                  fullWidth
                />
              </Col>
              <IfPermission perm="ui-organizations.creds.manage">
                <Col xs={12} md={6}>
                  <Field
                    label={<FormattedMessage id="ui-organizations.interface.username" />}
                    name="username"
                    component={TextField}
                    fullWidth
                  />
                </Col>
                <Col xs={12} md={6}>
                  <TogglePassword
                    name="password"
                    buttonID="button"
                  />
                </Col>
              </IfPermission>
              <Col xs={12}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.notes" />}
                  name="notes"
                  component={TextArea}
                  fullWidth
                />
              </Col>
              <Col xs={12}>
                <Headline
                  margin="none"
                  size="large"
                >
                  <FormattedMessage id="ui-organizations.interface.statistics" />
                </Headline>
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.available" />}
                  name="available"
                  component={Checkbox}
                  type="checkbox"
                />
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.deliveryMethod" />}
                  name="deliveryMethod"
                  component={Select}
                  fullWidth
                  dataOptions={deliveryMethodDD}
                />
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.statisticsFormat" />}
                  name="statisticsFormat"
                  component={Select}
                  fullWidth
                  dataOptions={formatDD}
                />
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.locallyStored" />}
                  name="locallyStored"
                  component={TextField}
                  fullWidth
                />
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.onlineLocation" />}
                  name="onlineLocation"
                  component={TextField}
                  fullWidth
                />
              </Col>
              <Col xs={12} md={6}>
                <Field
                  label={<FormattedMessage id="ui-organizations.interface.statisticsNotes" />}
                  name="statisticsNotes"
                  component={TextArea}
                  fullWidth
                />
              </Col>
            </Row>
          </Col>
        </Row>
      </Pane>
    </HasCommand>
  );
};

EditInterface.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  paneTitle: PropTypes.node.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  formatDD: PropTypes.arrayOf(PropTypes.object),
  deliveryMethodDD: PropTypes.arrayOf(PropTypes.object),
};

export default stripesForm({
  enableReinitialize: true,
  keepDirtyOnReinitialize: true,
  navigationCheck: true,
})(EditInterface);
