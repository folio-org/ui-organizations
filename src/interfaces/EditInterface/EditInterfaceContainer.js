import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';

import { get } from 'lodash';

import {
  interfaceCredentialsResource,
  interfaceResource,
  organizationResource,
} from '../../common/resources';
import { saveInterface } from './util';
import EditInterface from './EditInterface';
import {
  deliveryMethodDD,
  formatDD,
} from './const';
import { getBackPath } from '../../common/utils/createItem';

class EditInterfaceContainer extends Component {
  static manifest = Object.freeze({
    interfaceCredentials: interfaceCredentialsResource,
    interfaceId: {},
    interfaceOrg: organizationResource,
    vendorInterface: interfaceResource,
  });

  getCreds() {
    const { interfaceCredentials } = this.props.resources || {};

    return (!get(interfaceCredentials, 'failed') && get(interfaceCredentials, 'records.0')) || {};
  }

  onClose = (interfaceId = this.props.match.params.id) => {
    const { orgId, history } = this.props;

    history.push(getBackPath(orgId, interfaceId, 'interface'));
  };

  onSubmit = (formValues) => {
    const { mutator, resources, showMessage } = this.props;
    const creds = this.getCreds();

    saveInterface(mutator, formValues, creds, get(resources, 'interfaceOrg.records.0'))
      .then((id) => {
        showMessage('ui-organizations.interface.message.saved.success', 'success');
        this.onClose(id);
      })
      .catch(() => showMessage('ui-organizations.interface.message.saved.fail', 'error'));
  }

  render() {
    const { match, resources, stripes } = this.props;
    const isNew = !match.params.id;
    const loadedInterface = get(resources, 'vendorInterface.records[0]', {});
    const { username, password } = this.getCreds();
    const initialValues = isNew ? {} : {
      ...loadedInterface,
      username,
      password,
    };
    const { name } = initialValues;
    const paneTitle = isNew
      ? <FormattedMessage id="ui-organizations.interface.create.paneTitle" />
      : <FormattedMessage id="ui-organizations.interface.edit.paneTitle" values={{ name }} />;

    return (
      <EditInterface
        deliveryMethodDD={deliveryMethodDD}
        formatDD={formatDD}
        initialValues={initialValues}
        onClose={this.onClose}
        onSubmit={this.onSubmit}
        paneTitle={paneTitle}
        stripes={stripes}
      />
    );
  }
}

EditInterfaceContainer.propTypes = {
  mutator: PropTypes.object,
  orgId: PropTypes.string,
  resources: PropTypes.object,
  showMessage: PropTypes.func,
  stripes: PropTypes.object,
  match: ReactRouterPropTypes.match.isRequired,
  history: ReactRouterPropTypes.history.isRequired,
};

export default EditInterfaceContainer;
