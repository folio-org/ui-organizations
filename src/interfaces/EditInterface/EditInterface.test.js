import React from 'react';
import { render } from '@folio/jest-config-stripes/testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { useHistory } from 'react-router';

import {
  HasCommand,
} from '@folio/stripes/components';

import { orgInterface } from 'fixtures';

import { ORGANIZATIONS_ROUTE } from '../../common/constants';
import EditInterface from './EditInterface';

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: jest.fn(),
}));
jest.mock('@folio/stripes-components/lib/Commander', () => ({
  HasCommand: jest.fn(({ children }) => <div>{children}</div>),
}));

const defaultProps = {
  onSubmit: jest.fn(),
  initialValues: orgInterface,
  onClose: jest.fn(),
  paneTitle: orgInterface.firstName,
};
const renderEditInterface = (props = defaultProps) => render(
  <EditInterface {...props} />,
  { wrapper: MemoryRouter },
);

describe('EditContact', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct form structure', () => {
    const { asFragment } = renderEditInterface();

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Shortcuts', () => {
    beforeEach(() => {
      HasCommand.mockClear();
    });

    it('should cancel form when cancel shortcut is called', () => {
      renderEditInterface();
      HasCommand.mock.calls[0][0].commands.find(c => c.name === 'cancel').handler();

      expect(defaultProps.onClose).toHaveBeenCalled();
    });

    it('should navigate to list view when search shortcut is called', () => {
      const pushMock = jest.fn();

      useHistory.mockClear().mockReturnValue({
        push: pushMock,
      });

      renderEditInterface();
      HasCommand.mock.calls[0][0].commands.find(c => c.name === 'search').handler();

      expect(pushMock).toHaveBeenCalledWith(ORGANIZATIONS_ROUTE);
    });
  });
});
