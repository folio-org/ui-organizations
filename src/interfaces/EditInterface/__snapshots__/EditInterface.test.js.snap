// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditContact should render correct form structure 1`] = `
<DocumentFragment>
  <div>
    <section
      class="pane focusIndicator"
      id="edit-interface"
    >
      <div
        aria-labelledby="paneHeaderedit-interface-pane-title"
        class="paneHeader"
        data-test-pane-header="true"
        id="paneHeaderedit-interface"
        role="region"
        tabindex="-1"
      >
        <div
          class="paneHeaderButtonsArea first"
        >
          <div
            class="paneMenu"
          >
            <button
              aria-label="stripes-components.closeItem"
              class="iconButton medium paneHeaderCloseIcon PaneHeaderIconButton paneHeaderIconButton"
              data-test-pane-header-dismiss-button="true"
              icon="times"
              type="button"
            >
              <span
                class="iconButtonInner mediumOffset PaneHeaderIconButton__inner"
                data-test-pane-header-dismiss-button="true"
              >
                <span>
                  Icon
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="paneHeaderCenter"
        >
          <div
            class="paneHeaderCenterInner"
            id="paneHeaderedit-interface-pane-title"
          />
        </div>
        <div
          class="paneHeaderButtonsArea last"
        >
          <div
            class="paneMenu"
          >
            <button
              class="button primary marginBottom0"
              disabled=""
              title="ui-organizations.interface.button.save"
              type="submit"
            >
              <span
                class="inner"
              >
                ui-organizations.interface.button.save
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="paneContent hasPadding"
        id="edit-interface-content"
        tabindex="-1"
      >
        <div
          class="row"
        >
          <div
            class="col-xs-12 col-md-8 col-md-offset-2"
          >
            <div
              class="row"
            >
              <div
                class="col-xs-6"
              >
                <div
                  class="select inputGroup fullWidth"
                >
                  <label
                    class="label"
                    for="select-2"
                    id="select-2-label"
                  >
                    ui-organizations.interface.type
                  </label>
                  <div
                    class="selectWrap"
                  >
                    <select
                      aria-invalid="false"
                      aria-labelledby="select-2-label"
                      class="selectControl formControl"
                      id="select-2"
                      multiple=""
                      name="type"
                      type="select"
                    >
                      <option
                        value="Admin"
                      >
                        Admin
                      </option>
                      <option
                        value="End user"
                      >
                        End user
                      </option>
                      <option
                        value="Reports"
                      >
                        Reports
                      </option>
                      <option
                        value="Orders"
                      >
                        Orders
                      </option>
                      <option
                        value="Invoices"
                      >
                        Invoices
                      </option>
                      <option
                        value="Other"
                      >
                        Other
                      </option>
                    </select>
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="textField"
                >
                  <label
                    class="label"
                    for="text-input-3"
                  >
                    ui-organizations.interface.name
                    <span
                      aria-hidden="true"
                      class="asterisk"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="formControl inputGroup"
                  >
                    <input
                      aria-invalid="false"
                      aria-required="true"
                      autocapitalize="off"
                      autocomplete="off"
                      autocorrect="off"
                      error=""
                      id="text-input-3"
                      name="name"
                      required=""
                      spellcheck="false"
                      type="text"
                      value="GOBI"
                      warning=""
                    />
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="textField"
                >
                  <label
                    class="label"
                    for="text-input-4"
                  >
                    ui-organizations.interface.uri
                  </label>
                  <div
                    class="formControl inputGroup"
                  >
                    <input
                      aria-invalid="false"
                      autocapitalize="off"
                      autocomplete="off"
                      autocorrect="off"
                      error=""
                      id="text-input-4"
                      name="uri"
                      spellcheck="false"
                      type="url"
                      value="www.gobi3.com"
                      warning=""
                    />
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="textField"
                >
                  <label
                    class="label"
                    for="text-input-5"
                  >
                    ui-organizations.interface.username
                  </label>
                  <div
                    class="formControl inputGroup"
                  >
                    <input
                      aria-invalid="false"
                      autocapitalize="off"
                      autocomplete="off"
                      autocorrect="off"
                      error=""
                      id="text-input-5"
                      name="username"
                      spellcheck="false"
                      type="text"
                      value=""
                      warning=""
                    />
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-xs-10"
                  >
                    <div
                      class="textField"
                    >
                      <label
                        class="label"
                        for="text-input-6"
                      >
                        ui-organizations.edit.password
                      </label>
                      <div
                        class="formControl inputGroup"
                      >
                        <input
                          aria-invalid="false"
                          autocapitalize="off"
                          autocomplete="new-password"
                          autocorrect="off"
                          data-testid="password-field"
                          error=""
                          id="text-input-6"
                          name="password"
                          spellcheck="false"
                          type="password"
                          value=""
                          warning=""
                        />
                      </div>
                      <div
                        role="alert"
                      />
                    </div>
                  </div>
                  <div
                    class="col-xs-2"
                    style="padding-top: 20px; margin-bottom: 0px;"
                  >
                    <button
                      class="button default"
                      id="button"
                      type="button"
                    >
                      <span
                        class="inner"
                      >
                        ui-organizations.edit.show
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="col-xs-12"
              >
                <div
                  class="textArea fullWidth"
                >
                  <label
                    class="label"
                    for="textarea-input-7"
                  >
                    ui-organizations.interface.notes
                  </label>
                  <div
                    class="inputGroup"
                  >
                    <textarea
                      aria-invalid="false"
                      class="formControl"
                      id="textarea-input-7"
                      name="notes"
                    >
                      all staff have different logons with different permissions
                    </textarea>
                    <div
                      role="alert"
                    />
                  </div>
                </div>
              </div>
              <div
                class="col-xs-12"
              >
                <div
                  class="headline size-large margin-none font-weight-bold"
                  data-test-headline="true"
                >
                  ui-organizations.interface.statistics
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="checkbox"
                  data-test-checkbox="true"
                >
                  <label
                    class="label label checkboxInteractionStylesControl"
                    for="checkbox-8"
                  >
                    <span
                      class="inner checkboxInteractionStyles"
                    >
                      <input
                        aria-invalid="false"
                        class="input"
                        id="checkbox-8"
                        name="available"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="checkboxIcon"
                      >
                        <svg
                          height="10"
                          viewBox="0 0 14 14"
                          width="10"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M10.395 4.617L5.63 9.383 3.605 7.358"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-miterlimit="10"
                            stroke-width="1.5"
                          />
                        </svg>
                      </span>
                      <span
                        class="labelText"
                      >
                        ui-organizations.interface.available
                      </span>
                    </span>
                  </label>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="select inputGroup fullWidth"
                >
                  <label
                    class="label"
                    for="select-9"
                    id="select-9-label"
                  >
                    ui-organizations.interface.deliveryMethod
                  </label>
                  <div
                    class="selectWrap"
                  >
                    <select
                      aria-invalid="false"
                      aria-labelledby="select-9-label"
                      class="selectControl formControl"
                      id="select-9"
                      name="deliveryMethod"
                    />
                    <span>
                      Icon
                    </span>
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="select inputGroup fullWidth"
                >
                  <label
                    class="label"
                    for="select-10"
                    id="select-10-label"
                  >
                    ui-organizations.interface.statisticsFormat
                  </label>
                  <div
                    class="selectWrap"
                  >
                    <select
                      aria-invalid="false"
                      aria-labelledby="select-10-label"
                      class="selectControl formControl"
                      id="select-10"
                      name="statisticsFormat"
                    />
                    <span>
                      Icon
                    </span>
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="textField"
                >
                  <label
                    class="label"
                    for="text-input-11"
                  >
                    ui-organizations.interface.locallyStored
                  </label>
                  <div
                    class="formControl inputGroup"
                  >
                    <input
                      aria-invalid="false"
                      autocapitalize="off"
                      autocomplete="off"
                      autocorrect="off"
                      error=""
                      id="text-input-11"
                      name="locallyStored"
                      spellcheck="false"
                      type="text"
                      value="false"
                      warning=""
                    />
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="textField"
                >
                  <label
                    class="label"
                    for="text-input-12"
                  >
                    ui-organizations.interface.onlineLocation
                  </label>
                  <div
                    class="formControl inputGroup"
                  >
                    <input
                      aria-invalid="false"
                      autocapitalize="off"
                      autocomplete="off"
                      autocorrect="off"
                      error=""
                      id="text-input-12"
                      name="onlineLocation"
                      spellcheck="false"
                      type="text"
                      value=""
                      warning=""
                    />
                  </div>
                  <div
                    role="alert"
                  />
                </div>
              </div>
              <div
                class="col-xs-12 col-md-6"
              >
                <div
                  class="textArea fullWidth"
                >
                  <label
                    class="label"
                    for="textarea-input-13"
                  >
                    ui-organizations.interface.statisticsNotes
                  </label>
                  <div
                    class="inputGroup"
                  >
                    <textarea
                      aria-invalid="false"
                      class="formControl"
                      id="textarea-input-13"
                      name="statisticsNotes"
                    >
                      get stats on demand from GOBI, or contact customers service
                    </textarea>
                    <div
                      role="alert"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div />
</DocumentFragment>
`;
